eStore/Core/UI/Controls/SectionedMenuView.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  SectionedMenuView.swift
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 4/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import UIKit
+
+enum SectionedMenuRowType {
+    case thumbnail
+    case regular
+    case colorPicker
+}
+
+struct SectionedMenuSection {
+    let title: String?
+    let items: [SectionedMenuRow]
+}
+
+struct SectionedMenuRow {
+    let imageUrl: String?
+    let title: String?
+    let type: SectionedMenuRowType
+    let action: ((_ point: CGPoint) -> Void)?
+
+    init(imageUrl: String?,
+         title: String?,
+         type: SectionedMenuRowType = .regular,
+         action: ((_ point: CGPoint) -> Void)?) {
+
+        self.imageUrl = imageUrl
+        self.title = title
+        self.type = type
+        self.action = action
+    }
+}
+
+final class SectionedMenuView: UITableView {
+    var theme: Theme!
+    var sections: [SectionedMenuSection] = []
+
+    override init(frame: CGRect, style: UITableView.Style) {
+        super.init(frame: frame, style: style)
+
+        register(SectionedMenuRegularCell.self, forCellReuseIdentifier: SectionedMenuRegularCell.reuseIdentifier)
```

Comment:
I have some concerns about generic table view registration and reuse. Ask me IRL about that... I can share some snippets. They are pretty convenient to use.